custom_components/ha-google-home/sensor.py

```diff
@@ -46,46 +60,62 @@ def device_state_attributes(self):
 class GoogleHomeAlarmSensor(GoogleHomeAlarmEntity):
     """Representation of a Sensor."""
 
-    def __init__(self, coordinator, entry, device):
+    def __init__(self, coordinator, entry, device_name, alarms):
         """Initialize the sensor."""
         super().__init__(coordinator, entry)
-        self._name = device.device_name
-        self._alarms = [format_alarm_information(alarm) for alarm in device.alarms]
+        self._name = device_name
 
-        if len(self._alarms) > 0:
-            self._state = self._alarms[0][FIRE_TIME_IN_S]
-        else:
-            self._state = STATE_OFF
+    @property
+    def state(self):
+        alarms = self.get_alarms_data()
+        state = alarms[0][LOCAL_TIME] if alarms else STATE_OFF
+        return state
 
     @property
     def device_state_attributes(self):
         """Return the state attributes."""
         return {
-            "alarms": self._alarms,
+            "alarms": self.get_alarms_data(),
             "device": str(self.name),
             "integration": DOMAIN,
         }
 
+    def get_alarms_data(self):
+        """Update alarms data extracting it from coordinator"""
+        alarms = [
+            format_alarm_information(alarm)
+            for alarm in self.coordinator.data[self._name][LABEL_ALARMS]
+        ]
+        return alarms
+
 
 class GoogleHomeTimerSensor(GoogleHomeTimersEntity):
     """Representation of a Sensor."""
 
-    def __init__(self, coordinator, entry, device):
+    def __init__(self, coordinator, entry, device_name, timers):
         """Initialize the sensor."""
         super().__init__(coordinator, entry)
-        self._name = device.device_name
-        self._timers = [format_timer_information(timer) for timer in device.timers]
+        self._name = device_name
```

Comment:
In fact `timers` is never used (`pylint` or `flake8` should warn about this, probably worth setting them up) and you can remove it completely. I believe `entry` is also not used in entity, can be stripped off as well.

Reply:
Hm, weird that `flake8` didn't catch it, we have it installed into `pre-commit` hooks. Maybe that's because I use similarly named variables in the class. Thanks for spotting on those. I will leave it just for now, I believe there must be more work done further on sensor classes to implement proper data updating using `update` class (but I need to do a better lookup if that's true). So will keep these for now, but surely stripping them all down if not needed further down. :)